src/components/Pressable/Pressable.tsx

```diff
@@ -0,0 +1,279 @@
+import { GestureTouchEvent } from '../../handlers/gestureHandlerCommon';
+
+const DEFAULT_LONG_PRESS_DURATION = 500;
+
+export default function Pressable(props: PressableProps) {
+  const [pressedState, setPressedState] = useState(
+    props.testOnly_pressed ?? false
+  );
+
+  const pressableRef = useRef<View>(null);
+
+  // disabled when onLongPress has been called
+  const isPressCallbackEnabled = useRef<boolean>(true);
+  const isPressedDown = useRef<boolean>(false);
+
+  const normalizedHitSlop: Insets =
+    typeof props.hitSlop === 'number'
+      ? numberAsInset(props.hitSlop)
+      : props.hitSlop ?? {};
+
+  const normalizedPressRetentionOffset: Insets =
+    typeof props.pressRetentionOffset === 'number'
+      ? numberAsInset(props.pressRetentionOffset)
+      : props.pressRetentionOffset ?? {};
+
+  const pressGesture = useMemo(
+    () =>
+      Gesture.LongPress().onStart((event) => {
+        if (isPressedDown.current) {
+          props.onLongPress?.(adaptStateChangeEvent(event));
+          isPressCallbackEnabled.current = false;
+        }
+      }),
+    [isPressCallbackEnabled, props.onLongPress, isPressedDown]
+  );
+
+  const hoverInTimeout = useRef<number | null>(null);
+  const hoverOutTimeout = useRef<number | null>(null);
+
+  const hoverGesture = useMemo(
+    () =>
+      Gesture.Hover()
+        .onBegin((event) => {
+          if (hoverOutTimeout.current) {
+            clearTimeout(hoverOutTimeout.current);
+          }
+          if (props.delayHoverIn) {
+            hoverInTimeout.current = setTimeout(
+              () => props.onHoverIn?.(adaptStateChangeEvent(event)),
+              props.delayHoverIn
+            );
+            return;
+          }
+          props.onHoverIn?.(adaptStateChangeEvent(event));
+        })
+        .onEnd((event) => {
+          if (hoverInTimeout.current) {
+            clearTimeout(hoverInTimeout.current);
+          }
+          if (props.delayHoverOut) {
+            hoverOutTimeout.current = setTimeout(
+              () => props.onHoverOut?.(adaptStateChangeEvent(event)),
+              props.delayHoverOut
+            );
+            return;
+          }
+          props.onHoverOut?.(adaptStateChangeEvent(event));
+        }),
+    [props.onHoverIn, props.onHoverOut, props.delayHoverIn, props.delayHoverOut]
+  );
+
+  const pressDelayTimeoutRef = useRef<number | null>(null);
+  const pressInHandler = useCallback((event: GestureTouchEvent) => {
+    props.onPressIn?.(adaptTouchEvent(event));
+    isPressCallbackEnabled.current = true;
+    setPressedState(true);
+    pressDelayTimeoutRef.current = null;
+  }, []);
+  const pressOutHandler = useCallback((event: GestureTouchEvent) => {
+    if (
+      !isPressedDown.current ||
+      event.allTouches.length > event.changedTouches.length
+    ) {
+      return;
+    }
+
+    if (props.unstable_pressDelay && pressDelayTimeoutRef.current !== null) {
+      // if pressDelay is set, we want to call onPressIn on touch up
```

Comment:
It would be nice if you could also explain in that comment why we want to do this 😅

Reply:
This is a comment regarding how the legacy version behaved.
I changed it to:
```
      // legacy Pressable behaviour - if pressDelay is set, we want to call onPressIn on touch up
```

Reply:
What I meant was to explain why we want to call `onPressIn` on touch up, because it is not that obvious at first sight. We believe that the reason is because if you click on `Pressable` that has delay and then you lift the finger, you want to receive `pressIn` immediately (cc @j-piasecki). Adding that this is legacy behavior doesn't change much if you ask me.

Reply:
Replaced it with this:

![image](https://github.com/software-mansion/react-native-gesture-handler/assets/74246391/91ae95ca-c7f9-46b6-b960-f8ac570a5465)

I hope that plays a good role of explaining the launching of `pressInHandler` inside `pressOutHandler`

Reply:
It's fine. It is only a comment after all 😅